docs/content/integrations/bigquery/reference.mdx

```diff
@@ -0,0 +1,571 @@
+
+To avoid issues with newline characters in the GCP credential key, you must base64 encode the key. For example, if your GCP key is stored at `~.gcp/key.json` you can base64 encode the key by using the following shell command:
+
+```shell
+cat ~.gcp/key.json | base64
+```
+
+Then you can set an environment variable in your Dagster deployment (for example `GCP_CREDS`) to the encoded key and provide it to the BigQuery I/O Manager:
+
+```python file=/integrations/bigquery/config_auth.py startafter=start_example endbefore=end_example
+from dagster_gcp_pandas import bigquery_pandas_io_manager
+
+from dagster import Definitions
+
+defs = Definitions(
+    assets=[iris_data],
+    resources={
+        "io_manager": bigquery_pandas_io_manager.configured(
+            {
+                "project": "my-gcp-project",
+                "location": "us-east5",
+                "dataset": "IRIS",
+                "gcp_credential": {"env": "GCP_CREDS"},
+            }
+        )
+    },
+)
+```
+
+---
+
+## Selecting specific columns in a downstream asset
+
+Sometimes you may not want to fetch an entire table as the input to a downstream asset. With the BigQuery I/O manager, you can select specific columns to load by supplying metadata on the downstream asset.
+
+```python file=/integrations/bigquery/downstream_columns.py
+import pandas as pd
+
+from dagster import AssetIn, asset
+
+
+# this example uses the iris_data asset from Step 2 of the Using Dagster with BigQuery tutorial
+
+
+@asset(
+    ins={
+        "iris_sepal": AssetIn(
+            key="iris_data",
+            metadata={"columns": ["Sepal length (cm)", "Sepal width (cm)"]},
+        )
+    }
+)
+def sepal_data(iris_sepal: pd.DataFrame) -> pd.DataFrame:
+    iris_sepal["Sepal area (cm2)"] = (
+        iris_sepal["Sepal length (cm)"] * iris_sepal["Sepal width (cm)"]
+    )
+    return iris_sepal
+```
+
+In this example, we only use the columns containing sepal data from the `IRIS_DATA` table created in [Step 2: Create tables in BigQuery](/integrations/bigquery/using-bigquery-with-dagster#store-a-dagster-asset-as-a-table-in-bigquery) of the [Using Dagster with BigQuery tutorial](/integrations/bigquery/using-bigquery-with-dagster). Fetching the entire table would be unnecessarily costly, so to select specific columns, we can add metadata to the input asset. We do this in the `metadata` parameter of the `AssetIn` that loads the `iris_data` asset in the `ins` parameter. We supply the key `columns` with a list of names of the columns we want to fetch.
+
+When Dagster materializes `sepal_data` and loads the `iris_data` asset using the BigQuery I/O manager, it will only fetch the `Sepal length (cm)` and `Sepal width (cm)` columns of the `IRIS.IRIS_DATA` table and pass them to `sepal_data` as a Pandas DataFrame.
+
+---
+
+## Storing partitioned assets
```

Comment:
What would you think about putting the subsections in this section into tabs? ex:

```
## Storing partitioned assets

[intro as you have it]

<TabGroup>
<TabItem name="Static partitioned assets">
</TabItem>
<TabItem name="Time-partitioned assets">
</TabItem>
<TabItem name="Multi-partitioned assets">
</TabItem>
</TabGroup>
```

I realize I didn't suggest this for the Snowflake reference a few weeks ago, even though I thought I did. What are your thoughts?